### ISSUE TITLE
Tags: Unit Test hashing

### ISSUE THREAD

**Issue by eshurakov** [Automattic/simplenote-ios#1053]
Add unit tests for tag hashing function.

Use the following scenarios:

```
1.
Name: TáßvĒёи兔子@#$%^&*+-=_`~/?., ><{}[]\()"|':;
Hash: t%C3%A1%C3%9Fv%C4%93%D1%91%D0%B8%E5%85%94%E5%AD%90%40%23%24%25%5E%26%2A%2B%2D%3D%5F%60%7E%2F%3F%2E%2C%20%3E%3C%7B%7D%5B%5D%5C%28%29%22%7C%27%3A%3B

2. 
Name: \u00E9
Hash: %C3%A9

3.
Name: \u0065\u0301
Hash: %C3%A9

The names for 2 and 3 are the same (é), but represented differently and in the end should have the same hash.
```

Ref: p2XJRt-2tq

**Comment by charliescheer** [comment]
Hey @eshurakov I was looking into this and had a few questions for you.

First, the function that we are looking to test here is the String function byEncodingAsTagHash correct?

Also, for your first example.  The hash you provided here has an additional character that is not in the name.

`t%C3%A1%C3%9Fv%C4%93%D1%91%D0%B8%E5%85%94%E5%AD%90%40%23%24%25%5E%26%2A%2B%2D%3D%5F%60%7E%2F%3F%2E%2C%20%3E%3C%7B%7D%5B%5D_%5C_%28%29%22%7C%27%3A%3B`

That backslash character does not appear in the name that you used as your example.  Should the backslash be part of the name, or should the backslash not be in the hash output?

**Comment by eshurakov** [comment]
Hey @charliescheer! Good catch, `\` should be in the name, I missed it.

Correct, what we want to test is `byEncodingAsTagHash`, but I see that it already has some tests. It's good to check if we need to extend those tests, possibly they already cover everything from here. (I should've checked before creating this task 🙈)

Thank you!

**Comment by charliescheer** [comment]
Closed via #1096
